app/models/user.rb

```diff
@@ -1438,6 +1438,10 @@ def shelved_notifications
     ShelvedNotification.joins(:notification).where("notifications.user_id = ?", self.id)
   end
 
+  def ban_from_external_uploads!(ban_minutes: 5)
```

Comment:
I kind of feel like this doesn't belong in the `user` model. It is too specific to the `external_upload_manager` and it doesn't actually rely on the user record. We only need the `user`'s id for the Redis key. Perhaps `external_upload_manager` is the better place to extract this method to?

Reply:
Hmm yes I suppose this is fair, there shouldn't be anything else banning the user, at least for now. I have added a guardian check too, maybe I will do something like this:

```ruby
def can_upload_external?
  !ExternalUploadManager.user_banned?(@user)
end
```

Reply:
And just add a private method in `ExternalUploadManager` to ban the user :+1:

Reply:
Sounds good to me.